packages/@aws-cdk/pipelines/README.md

```diff
@@ -0,0 +1,542 @@
+    // load balancer address.
+    URL: pipeline.stackOutput(lbApp.loadBalancerAddress),
+  }
+});
+```
+
+### Using additional files in Shell Script Actions
+
+As part of a validation, you probably want to run a test suite that's more
+elaborate than what can be expressed in a couple of lines of shell script.
+You can bring additional files into the shell script validation by supplying
+the `additionalArtifacts` property.
+
+Here are some typical examples for how you might want to bring in additional
+files from several sources:
+
+* Directoy from the source repository
+* Additional compiled artifacts from the synth step
+
+#### Additional files from the source repository
+
+Bringing in additional files from the source repository is appropriate if the
+files in the source repository are directly usable in the test (for example,
+if they are executable shell scripts themselves). Pass the `sourceArtifact`:
+
+```ts
+const sourceArtifact = new codepipeline.Artifact();
+
+const pipeline = new CdkPipeline(this, 'Pipeline', {
+  // ...
+});
+
+const validationAction = new ShellScriptAction({
+  name: 'TestUsingSourceArtifact',
+  additionalArtifacts: [sourceArtifact],
+
+  // 'test.sh' comes from the source repository
+  commands: ['./test.sh'],
+});
+```
+
+#### Additional files from the synth step
+
+Getting the additional files from the synth step is appropriate if your
+tests need the compilation step that is done as part of synthesis.
+
+On the synthesis step, specify `additionalArtifacts` to package
+additional subdirectories into artifacts, and use the same artifact
+in the `ShellScriptAction`'s `additionalArtifacts`:
+
+```ts
+// If you are using additional output artifacts from the synth step,
+// they must be named.
+const cloudAssemblyArtifact = new codepipeline.Artifact('CloudAsm');
+const integTestsArtifact = new codepipeline.Artifact('IntegTests');
+
+const pipeline = new CdkPipeline(this, 'Pipeline', {
+  synthAction: SimpleSynthAction.standardNpmSynth({
+    sourceArtifact,
+    cloudAssemblyArtifact,
+    buildCommand: 'npm run build',
+    additionalArtifacts: [
+      {
+        directory: 'test',
+        artifact: integTestsArtifact,
+      }
+    ],
+  }),
+  // ...
+});
+
+const validationAction = new ShellScriptAction({
+  name: 'TestUsingBuildArtifact',
+  additionalArtifacts: [integTestsArtifact],
+  // 'test.js' was produced from 'test/test.ts' during the synth step
+  commands: ['node ./test.js'],
+});
+```
+
+## CDK Bootstrapping
+
+An *environment* is an *(account, region)* pair where you want to deploy a CDK
+stack (see [Environments](https://docs.aws.amazon.com/cdk/latest/guide/environments.html)
+in the CDK Developer Guide).
+
+Before you can provision the pipeline, you have to *bootstrap* the environment you want
+to create it in. If you are deploying your application to different environments, you
+also have to bootstrap those and be sure to add a *trust* relationship.
```

Comment:
Maybe add a preamble here which describes the scenario: a pipeline environment and an app environ...